pytest: add page (#9539)
tldr-pages/tldr: pages/common/pytest.md

# pytest

> Run Python tests.
> More information: <https://docs.pytest.org/>.

> - Run tests from a specific file:
> 
> `pytest {{path/to/test_file.py}}`

Reviewer: ```suggestion
- Run tests from specific files:

`pytest {{path/to/test_file1.py path/to/test_file2.py ...}}`
```

Optional change: several files possibly can be passed to conform command option [specification](https://manned.org/pytest.1): `python -m pytest [...]`.

> - Run all tests with a name that matches a given substring:
> 
> `pytest -k {{substring}}`

Reviewer: ```suggestion
- Run tests with names matching a specific [k]eyword expression:

`pytest -k {{expression}}`
```

Optional change: `all` can be assumed by default implicitly.
Optional change (no suggestions available based on man page): provide examples instead of generic `expression` placeholder delimited by `|` char.

> - Stop after the first test failure:

Author: ```suggestion
- Exit as soon as a test fails or encounters an error:
```

> - Stop after the first test failure:
> 
> `pytest --exitfirst`

Reviewer: ```suggestion
- Stop after a first test failure:

`pytest --exitfirst`
```

Optional change: `a` may sound better.

- Run tests matching or excluding markers:

`pytest -m {{marker_name1 and not marker_name2}}`

- Run until a test failure, continuing from the last failing test:

`pytest --stepwise`

> - Run tests without capturing output:
> 
> `pytest --capture=no`

Reviewer: ```suggestion
```

Optional fix: `--capture` is not mentioned in a [man](https://manned.org/pytest.1) page.
Note: possible bug in the corresponding man page.

Author: Usually the option is just `-s`, but I was following the style guide to use the long option.
